Expensify/App
# Update Plaid Link Component

### Details
Refactors this component as functional so we can use the Plaid Hooks. 

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/182956

### Tests
Same as QA done locally and android


### QA Steps
This should be tested on iOS
1. Create a new user and a workspace in NewDot.
2. Go to workspace settings and Connect a Bank
3. Follow the screenshots to go through the plaid oauth setup for PlatypusOAuth bank. You can enter empty fields for Purple OAuth forms like in the screenshots. 
4. Reach the final screenshot where you back to the expensify app

### Tested On


### Screenshots

<details>
  <summary>Screenshots take a lot of the screen so click here to expand/collapse them!</summary>

#### iOS
![image](https://user-images.githubusercontent.com/19364431/142699807-3d0089be-2204-4f8d-8ecb-babd1eaeeb42.png)

![image](https://user-images.githubusercontent.com/19364431/142699836-e0a44ce9-1247-41f1-b4b6-cb4ffb16f0f7.png)

![image](https://user-images.githubusercontent.com/19364431/142699893-a659bc8d-a18a-48d6-9a72-194054ed4b04.png)

![image](https://user-images.githubusercontent.com/19364431/142699912-46dcae94-7c23-425e-bbdc-113839b35a64.png)

![image](https://user-images.githubusercontent.com/19364431/142699928-5ca96729-3e17-4d39-98b7-58faf9ddea3a.png)

![image](https://user-images.githubusercontent.com/19364431/142699939-8625d87a-0f04-49e5-9201-252918f12972.png)

![image](https://user-images.githubusercontent.com/19364431/142699952-d83e94c9-cd37-482e-a998-93cf3d2e0a8b.png)

![image](https://user-images.githubusercontent.com/19364431/142700001-ace9eb63-5252-4ee5-b8fc-62eac8764c2e.png)

(scroll to the bottom)
![image](https://user-images.githubusercontent.com/19364431/142700016-39fc54ba-9f93-48e2-a373-6511e2174652.png)

![image](https://user-images.githubusercontent.com/19364431/142700063-42911c86-c4e5-4ef5-bddd-07a11566fcaa.png)

![image](https://user-images.githubusercontent.com/19364431/142700086-122b1d6a-abee-46bd-a419-13542a332784.png)

![image](https://user-images.githubusercontent.com/19364431/142700102-5d70237e-146e-4cd5-ba00-3c7a6fd7fab5.png)
  
</details>
